jabgui/src/test/java/org/jabref/gui/edit/CopyMoreActionTest.java

```diff
@@ -226,4 +247,298 @@ void executeCopyDoiOnSuccess() {
         verify(dialogService, times(1)).notify(Localization.lang("Copied '%0' to clipboard.",
                 JabRefDialogService.shortenDialogMessage(copiedDois)));
     }
+
+    @Test
+    void executeCopyAuthorWithNoAuthor() {
+        BibEntry entryWithNoAuthor = (BibEntry) entry.clone();
+        entryWithNoAuthor.clearField(StandardField.AUTHOR);
+        ObservableList<BibEntry> entriesWithNoAuthors = FXCollections.observableArrayList(entryWithNoAuthor);
+        BibDatabaseContext databaseContext = new BibDatabaseContext(new BibDatabase(entriesWithNoAuthors));
+
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.ofNullable(databaseContext));
+        when(stateManager.getSelectedEntries()).thenReturn(entriesWithNoAuthors);
+        copyMoreAction = new CopyMoreAction(StandardActions.COPY_FIELD_AUTHOR, dialogService, stateManager, clipBoardManager, preferences, abbreviationRepository);
+        copyMoreAction.execute();
+
+        verify(clipBoardManager, times(0)).setContent(any(String.class));
+        verify(dialogService, times(1)).notify(Localization.lang("None of the selected entries have %0.", "Author"));
+    }
+
+    @Test
+    void executeCopyAuthorOnPartialSuccess() {
+        BibEntry entryWithNoAuthor = (BibEntry) entry.clone();
+        entryWithNoAuthor.clearField(StandardField.AUTHOR);
+        ObservableList<BibEntry> mixedEntries = FXCollections.observableArrayList(entryWithNoAuthor, entry);
+        BibDatabaseContext databaseContext = new BibDatabaseContext(new BibDatabase(mixedEntries));
+
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.ofNullable(databaseContext));
+        when(stateManager.getSelectedEntries()).thenReturn(mixedEntries);
+        copyMoreAction = new CopyMoreAction(StandardActions.COPY_FIELD_AUTHOR, dialogService, stateManager, clipBoardManager, preferences, abbreviationRepository);
+        copyMoreAction.execute();
+
+        StringJoiner authorsJoiner = new StringJoiner("\n");
+        for (String author : authors) {
+            authorsJoiner.add(author);
+        }
+        String copiedAuthors = authorsJoiner.toString();
+        verify(clipBoardManager, times(1)).setContent(copiedAuthors);
+        verify(dialogService, times(1)).notify(Localization.lang("Warning: %0 out of %1 entries have undefined %2.",
+                Integer.toString(mixedEntries.size() - authors.size()), Integer.toString(mixedEntries.size()), "Author"));
+    }
+
+    @Test
+    void executeCopyAuthorOnSuccess() {
```

Comment:
I suspect most of the tests can be parameterized. Use methodsource with Arguments.of
eg. SuggentionProvidersTest